tasks/Task.cpp

```diff
@@ -57,27 +57,22 @@ void Task::processIO()
 
 }
 
-
-bool Task::setLight_level_all(int32_t level)
-{
-    lamps.setPowerUpLightLevel(level);
-    usleep(100000);
-    lamps.setLightLevel(level);
-    usleep(100000);
-    std::vector<ledlamp_bowtech::LedLamp> ledlist = _led_list.get();
-    for (size_t i = 0; i < ledlist.size(); i++)
-    {
-        ledlist[i].power_up_light_level = level;
-        ledlist[i].light_level = level;
-    }
-    _led_list.set(ledlist);
-
-    return(ledlamp_bowtech::TaskBase::setLight_level_all(level));
-}
-
 void Task::updateHook()
 {
     TaskBase::updateHook();
+
+    std::vector<double> led_level;
+    if(_led_input.read(led_level))
```

Comment:
This is not how InputPort::read() is used. Please have a look at the Rock tutorials or ask someone ...